File: src/dataset/go/hamming_Exercism_go_thinkverse/gemini/GeminiFlash_hamming_v3.go
package hamming

import (
	"errors"
)

func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 0, errors.New("Cannot calculate Hamming distance between samples or non-equal length.")
	}

	count := 0
	lenA := len(a)

	for i := 0; i < lenA; i++ {
		if a[i] != b[i] {
			count++
		}
	}

	return count, nil
}

File: src/dataset/cpp/hamming_exercism-johnngugi/gemini/GeminiFlash_hamming_v3.cpp
#include "hamming.h"
#include <string>
#include <stdexcept>

using namespace std;

int hamming::compute(string a, string b) {
  if (a.size() != b.size()) {
    throw domain_error("Error");
  }

  int distance = 0;
  size_t size = a.size();
  for (size_t i = 0; i < size; ++i) {
    if (a[i] != b[i]) {
      ++distance;
    }
  }
  return distance;
}